### ISSUE TITLE
[core] Improve documentation for Suppression

### ISSUE THREAD

**Issue by d1ss0nanz** [pmd/pmd#893]
**Affects PMD Version:** 
6.0.1

**Rule:**
Documenation. No specific rule.

**Description:**
The use of Violation Suppress Regex and Violation Suppress XPath needs better examples.

For the regex, it's not clear which message has to be matched (html?, xml?, what delimits "a" message?).

For XPath, a tutorial of how to use the "designer" would be great, plus a few real live examples.
And an explanation where in the AST the rule will match, so that one knows where a construct like ".[typeof(...)]" matches.

**Comment by d1ss0nanz** [comment]
Documentation was better in earlier versions:
https://pmd.github.io/pmd-5.4.0/customizing/howtowritearule.html

**Comment by rsoesemann** [comment]
Wow I didn't know of such a great documentation on writing rules. Is it in the 6.0 site as well?

**Comment by adangel** [comment]
Yes -> https://pmd.github.io/pmd-6.0.1/pmd_devdocs_writing_pmd_rules.html

**Comment by d1ss0nanz** [comment]
Sorry, didn't find it. m)
